bellboy/sensors.py

```diff
@@ -0,0 +1,145 @@
+# Sensor classes
+# Sensors include Camera, Ultrasonicm Mic....
+import logging
+import time
+from threading import Thread
+
+import picamera
+import RPi.GPIO as GPIO
+
+logger = logging.getLogger(__name__)
+
+
+class AbstractSensor():
+
+    # constructor.
+    # creates private variables for:
+    #   the sensor's data buffer,
+    #   the polling period, and
+    #   the sensor's polling thread
+    # @param poll_period_us time inbetween sensor reads, in microseconds
+    def __init__(self, poll_period_us):
+
+        self._poll_period = poll_period_us
+        self._buffer = []
+        super().__init__()
+        self._sensor_thread = Thread(target=self._read_sensor_loop)
+
+    # setup the sensor
+    def init_sensor(self):
+        pass
+
+    # close the sensor
+    def close_sensor(self):
+        pass
+
+    # string rep of sensors current status
+    def brief_status(self):
+        pass
+
+    # loop to be run in sensor's thread
+    def _read_sensor_loop(self):
+        # wait for self._poll_period
+        pass
+
+    def join(self):
+        self._sensor_thread.join()
+
+    def begin(self):
+        logger.debug("starting ultrasonic sensor thread")
+        self._sensor_thread.start()
+
+
+class TestSensor(AbstractSensor):
+    def __init__(self):
+        super().__init__(300)
+
+
+class RPICamSensor(AbstractSensor):
+    def __init__(self, poll_period_us):
+        super().__init__(poll_period_us)
+        self.picam: PiCamera = PiCamera()
+
+    def init_sensor(self):
+        pass
+
+    def close_sensor(self):
+        pass
+
+    def brief_status(self):
+        pass
+
+    def read_sensor(self):
+        pass
+
+
+class UltrasonicSensor(AbstractSensor):
+    def __init__(self, poll_period_us):
+        super().__init__(poll_period_us)
+
+    # setup the sensor
+    def init_sensor(self, trigPin=16, echoPin=18, max_depth_cm=200, pulse_width_us=10):
+        self._trigPin = trigPin
+        self._echoPin = echoPin
+        self._max_depth_cm = max_depth_cm
+        self._pulse_width = pulse_width_us
+        self._time_out = max_depth_cm*60*0.000001
+
+        GPIO.setup(trigPin, GPIO.OUT)   # set trigPin to OUTPUT mode
+        GPIO.setup(echoPin, GPIO.IN)    # set echoPin to INPUT mode
+
+    # close the sensor
+
+    def close_sensor(self):
+        GPIO.cleanup()         # release GPIO resource
+
+    # string rep of sensors current status
+    def brief_status(self):
+        pass
+
+    # read sensor once.
+    # to be called continuously in sensor's thread
+    def _read_sensor_loop(self):
+        # send high pulse
+        global logger
```

Comment:
Codacy found an issue: [Using the global statement](https://app.codacy.com/gh/Bellboy-Capstone/System/pullRequest?prid=6272175)